tests/functional/adapter/test_concurrency.py

```diff
@@ -0,0 +1,224 @@
+import pytest
+from dbt.tests.adapter.concurrency.test_concurrency import (
+    BaseConcurrency,
+)
+from dbt.tests.util import (
+    check_relations_equal,
+    check_table_does_not_exist,
+    rm_file,
+    run_dbt,
+    run_dbt_and_capture,
+    write_file,
+)
+from tests.fixtures.profiles import unique_schema, dbt_profile_data
+
+models__invalid_sql = """
+{{
+  config(
+    materialized = "table"
+  )
+}}
+
+select a_field_that_does_not_exist from {{ ref(var('seed_name', 'seed')) }}
+
+"""
+
+models__table_a_sql = """
+{{
+  config(
+    materialized = "table"
+  )
+}}
+
+select * from {{ ref(var('seed_name', 'seed')) }}
+
+"""
+
+models__table_b_sql = """
+{{
+  config(
+    materialized = "table"
+  )
+}}
+
+select * from {{ ref(var('seed_name', 'seed')) }}
+
+"""
+
+models__view_model_sql = """
+{{
+  config(
+    materialized = "view"
+  )
+}}
+
+select * from {{ ref(var('seed_name', 'seed')) }}
+
+"""
+
+models__dep_sql = """
+{{
+  config(
+    materialized = "table"
+  )
+}}
+
+select * from {{ref('view_model')}}
+
+"""
+
+models__view_with_conflicting_cascade_sql = """
+select * from {{ref('table_a')}}
+
+union all
+
+select * from {{ref('table_b')}}
+
+"""
+
+models__skip_sql = """
+select * from {{ref('invalid')}}
+
+"""
+
```

Comment:
For models/seeds that you didn't make changes to, you can just import from the appropriate file in dbt-core.